client/layout/index.jsx

```diff
@@ -114,6 +115,32 @@ class Layout extends Component {
 		// intentionally don't remove these in unmount
 	}
 
+	shouldShowHappyChatButton() {
+		if ( ! config.isEnabled( 'happychat' ) ) {
+			return false;
+		}
+
+		if ( isWpMobileApp() ) {
+			return false;
+		}
+
+		if ( ! this.props.hasActiveHappyChat ) {
+			return false;
+		}
+
+		const exemptedSections = [ 'jetpack-connect', 'happychat', 'devdocs' ];
```

Comment:
It makes sense to hide the Inline Help FAB on pages like [Jetpack Connect](https://wordpress.com/jetpack/connect) (since it has its own Help link with a different contact flow) — but can we be looser with restricting where the Happy Chat button is shown? I'm imagining that it's possible a WPCOM HE could be guiding a user through Jetpack connection, and it would actually be _useful_ to open up Happy Chat in the Jetpack Connect flow. Or maybe down the line Jetpack does Chat again and it takes someone hours to figure out why an existing Chat can't be opened on these pages...

I guess my thought is, since that button only shows up when the customer already knows they're chatting with someone, maybe we can avoid some of this logic (and possible future debug) by losing some of these restrictions.

I'm not strongly opinionated here, just offering as a thought.

Reply:
👋 @mattwondra! It seems that enabling the chat button in [Jetpack connect](https://wordpress.com/jetpack/connect) is a good strategy, thanks for pointing that out!

2e45b6f